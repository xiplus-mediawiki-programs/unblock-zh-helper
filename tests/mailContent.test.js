import { mount } from '@vue/test-utils';
import App from '../src/App.vue';
import createMyI18n from '../src/lang';
import { expect, test, describe } from 'vitest';

describe('test variant', async () => {
  const wrapper = mount(App, {
    global: {
      plugins: [createMyI18n('zh-hant')],
    },
  });
  await wrapper.setData({
    mailOptionsVariant: 'zh-hans',
  });

  test('hans', async () => {
    await wrapper.vm.resetForm();
    expect(wrapper.vm.mailContent).toMatchSnapshot();
  });
});

describe('mail content', async () => {
  const wrapper = mount(App, {
    global: {
      plugins: [createMyI18n('zh-hant')],
    },
  });
  await wrapper.setData({
    mailOptionsVariant: 'zh-hant',
  });

  test('init', async () => {
    await wrapper.vm.resetForm();
    expect(wrapper.vm.mailContent).toMatchSnapshot();
  });

  test('req account & ipbe, done', async () => {
    await wrapper.vm.resetForm();
    await wrapper.setData({
      mailOptionsUsername: wrapper.vm.MAILOP_ACCOUNTCREATED,
      mailOptionsIpbe: wrapper.vm.MAILOP_IPBEGRANTED,
      normalizedUsername: 'ExampleUser',
      email: 'email@example.org',
    });
    expect(wrapper.vm.mailContentCore).toMatchSnapshot();
  });

  test('req account & ipbe, no username', async () => {
    await wrapper.vm.resetForm();
    await wrapper.setData({ mailOptionsUsername: wrapper.vm.MAILOP_NOUSERNAME });
    expect(wrapper.vm.mailContentCore).toMatchSnapshot();
  });

  test('req account & ipbe, no username & ip', async () => {
    await wrapper.vm.resetForm();
    await wrapper.setData({
      mailOptionsUsername: wrapper.vm.MAILOP_NOUSERNAME,
      mailOptionsIpbe: wrapper.vm.MAILOP_NOIP,
    });
    expect(wrapper.vm.mailContentCore).toMatchSnapshot();
  });

  test('req account & ipbe, acct created, no ip', async () => {
    await wrapper.vm.resetForm();
    await wrapper.setData({
      mailOptionsUsername: wrapper.vm.MAILOP_ACCOUNTCREATED,
      normalizedUsername: 'ExampleUser',
      email: 'email@example.org',
    });
    expect(wrapper.vm.mailContentCore).toMatchSnapshot();
  });

  test('req account & ipbe, acct created, ip not blocked', async () => {
    await wrapper.vm.resetForm();
    await wrapper.setData({
      mailOptionsUsername: wrapper.vm.MAILOP_ACCOUNTCREATED,
      mailOptionsIpbe: wrapper.vm.MAILOP_IPNOTBLOCKED,
      normalizedUsername: 'ExampleUser',
      email: 'email@example.org',
    });
    expect(wrapper.vm.mailContentCore).toMatchSnapshot();
  });

  test('req ipbe, granted', async () => {
    await wrapper.vm.resetForm();
    await wrapper.setData({
      mailOptionsIpbe: wrapper.vm.MAILOP_IPBEGRANTED,
    });
    expect(wrapper.vm.mailContentCore).toMatchSnapshot();
  });

  test('req ipbe, no username', async () => {
    await wrapper.vm.resetForm();
    await wrapper.setData({
      inputCreateAccount: false,
      mailOptionsUsername: wrapper.vm.MAILOP_NOUSERNAME,
    });
    expect(wrapper.vm.mailContentCore).toMatchSnapshot();
  });

  test('username used', async () => {
    await wrapper.vm.resetForm();
    await wrapper.setData({
      mailOptionsUsername: wrapper.vm.MAILOP_USERNAMEUSED,
      usernameRegistration: '2019-01-12T12:34:56Z',
    });
    expect(wrapper.vm.mailContentCore).toMatchSnapshot();
  });

  test('username banned', async () => {
    await wrapper.vm.resetForm();
    await wrapper.setData({
      mailOptionsUsername: wrapper.vm.MAILOP_USERNAMEBANNED,
    });
    expect(wrapper.vm.mailContentCore).toMatchSnapshot();
  });

  test('username illeagal', async () => {
    await wrapper.vm.resetForm();
    await wrapper.setData({
      mailOptionsUsername: wrapper.vm.MAILOP_USERNAMEILLEAGAL,
    });
    expect(wrapper.vm.mailContentCore).toMatchSnapshot();
  });

  test('open proxy', async () => {
    await wrapper.vm.resetForm();
    await wrapper.setData({
      mailOptionsOther: wrapper.vm.MAILOP_OPENPROXY,
    });
    expect(wrapper.vm.mailContentCore).toMatchSnapshot();
  });

  test('range block', async () => {
    await wrapper.vm.resetForm();
    await wrapper.setData({
      mailOptionsOther: wrapper.vm.MAILOP_RANGEBLOCK,
    });
    expect(wrapper.vm.mailContentCore).toMatchSnapshot();
  });

  test('enwiki block', async () => {
    await wrapper.vm.resetForm();
    await wrapper.setData({
      mailOptionsOther: wrapper.vm.MAILOP_ENWIKIBLOCK,
    });
    expect(wrapper.vm.mailContentCore).toMatchSnapshot();
  });

  test('gipbe', async () => {
    await wrapper.vm.resetForm();
    await wrapper.setData({
      mailOptionsOther: wrapper.vm.MAILOP_GIPBE,
    });
    expect(wrapper.vm.mailContentCore).toMatchSnapshot();
  });

  test('company basic', async () => {
    await wrapper.vm.resetForm();
    await wrapper.setData({
      mailOptionsOther: wrapper.vm.MAILOP_COMPANY,
    });
    expect(wrapper.vm.mailContentCore).toMatchSnapshot();
  });

  test('company & no username & ip', async () => {
    await wrapper.vm.resetForm();
    await wrapper.setData({
      mailOptionsUsername: wrapper.vm.MAILOP_NOUSERNAME,
      mailOptionsIpbe: wrapper.vm.MAILOP_NOIP,
      mailOptionsOther: wrapper.vm.MAILOP_COMPANY,
    });
    expect(wrapper.vm.mailContentCore).toMatchSnapshot();
  });

  test('auto logout', async () => {
    await wrapper.vm.resetForm();
    await wrapper.setData({
      mailOptionsOther: wrapper.vm.MAILOP_AUTOLOGOUT,
    });
    expect(wrapper.vm.mailContentCore).toMatchSnapshot();
  });

  test('user rename', async () => {
    await wrapper.vm.resetForm();
    await wrapper.setData({
      mailOptionsOther: wrapper.vm.MAILOP_USERRENAME,
    });
    expect(wrapper.vm.mailContentCore).toMatchSnapshot();
  });

  test('go talkpage', async () => {
    await wrapper.vm.resetForm();
    await wrapper.setData({
      mailOptionsOther: wrapper.vm.MAILOP_GOTALKPAGE,
    });
    expect(wrapper.vm.mailContentCore).toMatchSnapshot();
  });

  test('talkpage requested', async () => {
    await wrapper.vm.resetForm();
    await wrapper.setData({
      mailOptionsOther: wrapper.vm.MAILOP_TALKPAGEREQED,
    });
    expect(wrapper.vm.mailContentCore).toMatchSnapshot();
  });

  test('can not understand', async () => {
    await wrapper.vm.resetForm();
    await wrapper.setData({
      mailOptionsOther: wrapper.vm.MAILOP_CANTUNDERSTAND,
    });
    expect(wrapper.vm.mailContentCore).toMatchSnapshot();
  });
});
