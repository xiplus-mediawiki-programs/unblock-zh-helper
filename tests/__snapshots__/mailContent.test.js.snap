// Vitest Snapshot v1

exports[`mail content > auto logout 1`] = `
"您好：
關於自動登出的問題，若您使用手機應用程式（App）會遇到這個問題，暫時無法解決，請改為使用網頁版。其他情況的解決方法可參見維基百科上的說明 [1]

[1] https://zh.wikipedia.org/wiki/Help:自動登出"
`;

exports[`mail content > can not understand 1`] = `
"您好：
請說明我們需要為您做什麼。您可以閱讀申訴指南 [1] 以了解我們通常需要您提供哪些資訊來處理您的問題。

[1] https://zh.wikipedia.org/wiki/Wikipedia:通过Unblock-zh申请IP封禁例外指南"
`;

exports[`mail content > company & no username & ip 1`] = `
"您好：
先為您簡要說明幾項維基百科的規定：
　1. 禁止多人共用帳號，禁止和其他人分享密碼，如果您的公司/組織有多個人需要編輯，請每一位都各自申請帳號。
　2. 禁止使用公司/組織的名稱作為使用者名稱，這個帳號代表您一個人，請為自己取個名稱 [1]。也請您提供您私人的電子郵件地址（非組織域名）來接受隨機密碼郵件。
　3. 如果是因為您在公司/組織的工作而來編輯維基百科，維基百科禁止您「直接」對頁面做出編輯，您必須：
　　3-1. 公開聲明您的僱用者、客戶、組織，具體操作請之後參閱Wikipedia:有償編輯方針#如何作出申報 [2]
　　3-2. 向其他人說明您想要編輯的內容並獲得同意，具體操作請之後參閱Wikipedia:有償編輯方針#本地替代方針 [3]
　4. 更多相關的說明請見Wikipedia:如何介绍自己的公司 [4] ，請您在編輯前務必閱讀並理解這篇說明，如果上述說明或給出連結的內容有任何不理解的地方，或您有任何其他問題，請在Wikipedia:互助客栈/求助 [5] 發問。
若您同意上述要求，我們仍可完成您的請求。

請告知以下資訊：
　1. 您想要的使用者名稱，「不要提供密碼」。請務必使用使用者名稱檢查工具[6]來確認您想要註冊的使用者名稱是否可用。
　2. 您私人的電子郵件地址（不接受組織域名的地址），用來接收隨機密碼及綁定到您的帳號。
　3. 被封鎖的IP位址
　4. 封鎖ID（如果有，可在告知被封鎖頁面看到「您目前的IP位址是xxxx，而該封鎖ID是#xxxx。」，這不是使用者名稱）
以便我們做下一步處理（複製您看到的文字對我們處理較為方便，請避免使用截圖）。

[1] https://zh.wikipedia.org/wiki/Wikipedia:用户名
[2] https://zh.wikipedia.org/wiki/Wikipedia:有償編輯方針#如何作出申報
[3] https://zh.wikipedia.org/wiki/Wikipedia:有償編輯方針#本地替代方針
[4] https://zh.wikipedia.org/wiki/Wikipedia:如何介绍自己的公司
[5] https://zh.wikipedia.org/wiki/Wikipedia:互助客栈/求助
[6] https://zhwiki-username-check.toolforge.org"
`;

exports[`mail content > company basic 1`] = `
"您好：
先為您簡要說明幾項維基百科的規定：
　1. 禁止多人共用帳號，禁止和其他人分享密碼，如果您的公司/組織有多個人需要編輯，請每一位都各自申請帳號。
　2. 禁止使用公司/組織的名稱作為使用者名稱，這個帳號代表您一個人，請為自己取個名稱 [1]。也請您提供您私人的電子郵件地址（非組織域名）來接受隨機密碼郵件。
　3. 如果是因為您在公司/組織的工作而來編輯維基百科，維基百科禁止您「直接」對頁面做出編輯，您必須：
　　3-1. 公開聲明您的僱用者、客戶、組織，具體操作請之後參閱Wikipedia:有償編輯方針#如何作出申報 [2]
　　3-2. 向其他人說明您想要編輯的內容並獲得同意，具體操作請之後參閱Wikipedia:有償編輯方針#本地替代方針 [3]
　4. 更多相關的說明請見Wikipedia:如何介绍自己的公司 [4] ，請您在編輯前務必閱讀並理解這篇說明，如果上述說明或給出連結的內容有任何不理解的地方，或您有任何其他問題，請在Wikipedia:互助客栈/求助 [5] 發問。
若您同意上述要求，我們仍可完成您的請求。

[1] https://zh.wikipedia.org/wiki/Wikipedia:用户名
[2] https://zh.wikipedia.org/wiki/Wikipedia:有償編輯方針#如何作出申報
[3] https://zh.wikipedia.org/wiki/Wikipedia:有償編輯方針#本地替代方針
[4] https://zh.wikipedia.org/wiki/Wikipedia:如何介绍自己的公司
[5] https://zh.wikipedia.org/wiki/Wikipedia:互助客栈/求助"
`;

exports[`mail content > enwiki block 1`] = `
"您好：
由於各個語言的維基百科是各自獨立管理的，這裡僅能處理中文維基百科[1]的問題，很抱歉幫不上忙。
英文維基百科上的申訴請自行向英文維基百科申請，請參考英文維基百科的說明[2]。

[1] https://zh.wikipedia.org
[2] https://en.wikipedia.org/wiki/Wikipedia:Unblock_Ticket_Request_System"
`;

exports[`mail content > gipbe 1`] = `
"您好：
由於各個語言的維基百科是各自獨立管理的，這裡僅能處理中文維基百科[1]的問題，很抱歉幫不上忙。
若有需要全域IP封鎖例外權，請參考Meta-Wiki上的說明[2]來申請。

[1] https://zh.wikipedia.org
[2] https://meta.wikimedia.org/wiki/IP_block_exempt/zh"
`;

exports[`mail content > go talkpage 1`] = `
"您好：
因為您未被禁止編輯您的討論頁，請在站內進行申訴，處理能較為快速。
請登入後點擊此連結進行申訴：[1]

[1] https://zh.wikipedia.org/w/index.php?title=Special:MyTalk&action=edit&preload=Template%3AUnblock%2Fpreload2&section=new"
`;

exports[`mail content > init 1`] = `
"您好：


請在回覆郵件時確保收件者包含 unblock-zh@lists.wikimedia.org ，否則可能無法收到回覆（可使用郵件軟體的「回覆所有人」功能）

User:Wikiuser"
`;

exports[`mail content > open proxy 1`] = `
"您好：
維基媒體基金會禁止使用某些開放代理來編輯維基百科[1]，所以我們不會解除封鎖這個IP，但我們可以為您的帳號授予豁免權限以繞過這個限制。

請告知您的使用者名稱（如果有，登入後從偏好設定查看[2]，這不是電子郵件位址）。
如果您沒有帳號且無法自行註冊，請告知您想要的使用者名稱，「不要提供密碼」。請務必使用使用者名稱檢查工具[3]來確認您想要註冊的使用者名稱是否可用。

[1] https://meta.wikimedia.org/wiki/No_open_proxies/zh
[2] https://zh.wikipedia.org/wiki/Special:Preferences
[3] https://zhwiki-username-check.toolforge.org"
`;

exports[`mail content > range block 1`] = `
"您好：
由於有其他人使用此段IP位址對維基百科進行破壞，所以您使用這一個範圍的IP位址已經被暫時封鎖。
如果您與破壞行為無關，請註冊一個帳號並登入後即可編輯。若您無法自行註冊帳號，請回信告知您想要的使用者名稱，「不要提供密碼」。請務必使用使用者名稱檢查工具[1]來確認您想要註冊的使用者名稱是否可用。

[1] https://zhwiki-username-check.toolforge.org"
`;

exports[`mail content > req account & ipbe, acct created, ip not blocked 1`] = `
"您好：
已代為註冊帳號「ExampleUser」，帳號的隨機密碼用另一封郵件寄出到「email@example.org」，隨機密碼的有效期限僅有7天，請盡速登入修改密碼。

您所給的IP位址未被封鎖，請確認正確的IP位址或封鎖ID後再回信，您可在告知被封鎖頁面看到「您目前的IP位址是xxxx，而該封鎖ID是#xxxx。」，若您已經可以編輯，則不用回信。"
`;

exports[`mail content > req account & ipbe, acct created, no ip 1`] = `
"您好：
已代為註冊帳號「ExampleUser」，帳號的隨機密碼用另一封郵件寄出到「email@example.org」，隨機密碼的有效期限僅有7天，請盡速登入修改密碼。"
`;

exports[`mail content > req account & ipbe, done 1`] = `
"您好：
已代為註冊帳號「ExampleUser」，帳號的隨機密碼用另一封郵件寄出到「email@example.org」，隨機密碼的有效期限僅有7天，請盡速登入修改密碼。
已授予您IP封鎖例外權限，登入後即可編輯頁面。祝您編輯愉快。"
`;

exports[`mail content > req account & ipbe, no username & ip 1`] = `
"您好：
請告知以下資訊：
　1. 您想要的使用者名稱，「不要提供密碼」。請務必使用使用者名稱檢查工具[1]來確認您想要註冊的使用者名稱是否可用。
　2. 被封鎖的IP位址
　3. 封鎖ID（如果有，可在告知被封鎖頁面看到「您目前的IP位址是xxxx，而該封鎖ID是#xxxx。」，這不是使用者名稱）
以便我們做下一步處理（複製您看到的文字對我們處理較為方便，請避免使用截圖）。

[1] https://zhwiki-username-check.toolforge.org"
`;

exports[`mail content > req account & ipbe, no username 1`] = `
"您好：
請告知您想要的使用者名稱，「不要提供密碼」。請務必使用使用者名稱檢查工具[1]來確認您想要註冊的使用者名稱是否可用。

[1] https://zhwiki-username-check.toolforge.org"
`;

exports[`mail content > req ipbe, granted 1`] = `
"您好：
已授予您IP封鎖例外權限，登入後即可編輯頁面。祝您編輯愉快。"
`;

exports[`mail content > req ipbe, no username 1`] = `
"您好：
請告知您的使用者名稱（如果有，登入後從偏好設定查看[1]，這不是電子郵件位址）。
如果您沒有帳號且無法自行註冊，請告知您想要的使用者名稱，「不要提供密碼」。請務必使用使用者名稱檢查工具[2]來確認您想要註冊的使用者名稱是否可用。

[1] https://zh.wikipedia.org/wiki/Special:Preferences
[2] https://zhwiki-username-check.toolforge.org"
`;

exports[`mail content > talkpage requested 1`] = `
"您好：
因為您已成功在站內發起申訴，這邊就不再重複處理，請您跟進站內的討論，謝謝。"
`;

exports[`mail content > user rename 1`] = `
"您好：
請您先至 Special:全域重命名申请 [1] 申請變更您的使用者名稱，待更名完成後再回信，我們將在您完成更名後再處理您的其他請求。

[1] https://zh.wikipedia.org/wiki/Special:全域重命名申请"
`;

exports[`mail content > username banned 1`] = `
"您好：
您所指定的使用者名稱被系統禁止。請提供另一個使用者名稱，請務必使用使用者名稱檢查工具[1]來確認您想要註冊的使用者名稱是否可用。

[1] https://zhwiki-username-check.toolforge.org"
`;

exports[`mail content > username illeagal 1`] = `
"您好：
您所指定的使用者名稱違反我們的規定，您不一定要使用本名，可使用暱稱，但使用者名稱應代表您自己。請閱讀使用者名稱方針 [1] 後提供另一個使用者名稱，請務必使用使用者名稱檢查工具[2]來確認您想要註冊的使用者名稱是否可用。

[1] https://zh.wikipedia.org/wiki/Wikipedia:用户名
[2] https://zhwiki-username-check.toolforge.org"
`;

exports[`mail content > username used 1`] = `
"您好：
您所指定的使用者名稱已經在 2019/1/12 被註冊 [1]，這是您嗎？若不是，請提供另一個使用者名稱，請務必使用使用者名稱檢查工具[2]來確認您想要註冊的使用者名稱是否可用。

[1] https://zh.wikipedia.org/w/index.php?title=Special:CentralAuth&target=ExampleUser
[2] https://zhwiki-username-check.toolforge.org"
`;

exports[`test variant > hans 1`] = `
"您好：


请在回复邮件时确保收件者包含 unblock-zh@lists.wikimedia.org ，否则可能无法收到回复（可使用邮件软件的“回复所有人”功能）

User:Wikiuser"
`;
